Build fails

Hi,

Following the instructions in README, I'm getting this error when running the `build.sh` script:

```
[ 72%] Building CXX object CMakeFiles/rellic-decomp-4.0.dir/rellic/AST/Util.cpp.o
/var/tmp/xchalup4/rellic-build/libraries/llvm/bin/clang++  -DGFLAGS_IS_A_DLL=0 -DGOOGLE_GLOG_DLL_DECL="" -DNDEBUG -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -isystem /var/tmp/xchalup4/rellic-build -isystem /var/tmp/xchalup4/rellic -isystem /var/tmp/xchalup4/rellic-build/libraries/llvm/include -isystem /var/tmp/xchalup4/rellic-build/libraries/z3/include -isystem /var/tmp/xchalup4/rellic-build/libraries/glog/include -isystem /var/tmp/xchalup4/rellic-build/libraries/gflags/include  -O2 -g -DNDEBUG   -Wall -Wextra -Wno-unused-parameter -Wno-c++98-compat -Wno-unreachable-code-return -Wno-nested-anon-types -Wno-extended-offsetof -Wno-variadic-macros -Wno-return-type-c-linkage -Wno-c99-extensions -Wno-ignored-attributes -Wno-unused-local-typedef -Wno-unknown-pragmas -Wno-unknown-warning-option -fPIC -fno-omit-frame-pointer -fvisibility-inlines-hidden -fno-exceptions -fno-asynchronous-unwind-tables -Wgnu-alignof-expression -Wno-gnu-anonymous-struct -Wno-gnu-designator -Wno-gnu-zero-variadic-macro-arguments -Wno-gnu-statement-expression -gdwarf-2 -g3 -O3 -Werror -pedantic -fopenmp=libomp -std=c++11 -o CMakeFiles/rellic-decomp-4.0.dir/rellic/AST/Util.cpp.o -c /var/tmp/xchalup4/rellic/rellic/AST/Util.cpp
/var/tmp/xchalup4/rellic/rellic/AST/IRToASTVisitor.cpp:335:31: error: no member named 'indices' in 'llvm::GetElementPtrInst'
    for (auto &gep_idx : inst.indices()) {
                         ~~~~ ^
1 error generated.
make[2]: *** [CMakeFiles/rellic-decomp-4.0.dir/build.make:157: CMakeFiles/rellic-decomp-4.0.dir/rellic/AST/IRToASTVisitor.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
/var/tmp/xchalup4/rellic/rellic/AST/ExprCombine.cpp:109:17: error: no matching function for call to 'ignoringParens'
            has(ignoringParens(binaryOperator(stmt().bind("binop")))))) {}
                ^~~~~~~~~~~~~~
/var/tmp/xchalup4/rellic-build/libraries/llvm/include/clang/ASTMatchers/ASTMatchers.h:728:25: note: candidate function not viable: no known conversion from
      'clang::ast_matchers::internal::BindableMatcher<clang::Stmt>' to 'const internal::Matcher<QualType>' for 1st argument
AST_MATCHER_P(QualType, ignoringParens,
                        ^
/var/tmp/xchalup4/rellic-build/libraries/llvm/include/clang/ASTMatchers/ASTMatchersMacros.h:130:32: note: expanded from macro 'AST_MATCHER_P'
  AST_MATCHER_P_OVERLOAD(Type, DefineMatcher, ParamType, Param, 0)
                               ^
/var/tmp/xchalup4/rellic-build/libraries/llvm/include/clang/ASTMatchers/ASTMatchersMacros.h:150:57: note: expanded from macro 'AST_MATCHER_P_OVERLOAD'
  inline ::clang::ast_matchers::internal::Matcher<Type> DefineMatcher(         \
                                                        ^
1 error generated.
make[2]: *** [CMakeFiles/rellic-decomp-4.0.dir/build.make:131: CMakeFiles/rellic-decomp-4.0.dir/rellic/AST/ExprCombine.cpp.o] Error 1
```

Btw. I tried to change the version of LLVM to llvm70 in the `build.sh` script and still getting the later error:

```
[ 72%] Building CXX object CMakeFiles/rellic-decomp-7.0.dir/rellic/AST/Util.cpp.o
/var/tmp/xchalup4/rellic/rellic-build/libraries/llvm/bin/clang++  -DGFLAGS_IS_A_DLL=0 -DGOOGLE_GLOG_DLL_DECL="" -DNDEBUG -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -isystem /var/tmp/xchalup4/rellic/rellic-build -isystem /var/tmp/xchalup4/rellic -isystem /var/tmp/xchalup4/rellic/rellic-build/libraries/llvm/include -isystem /var/tmp/xchalup4/rellic/rellic-build/libraries/z3/include -isystem /var/tmp/xchalup4/rellic/rellic-build/libraries/glog/include -isystem /var/tmp/xchalup4/rellic/rellic-build/libraries/gflags/include  -O2 -g -DNDEBUG   -Wall -Wextra -Wno-unused-parameter -Wno-c++98-compat -Wno-unreachable-code-return -Wno-nested-anon-types -Wno-extended-offsetof -Wno-variadic-macros -Wno-return-type-c-linkage -Wno-c99-extensions -Wno-ignored-attributes -Wno-unused-local-typedef -Wno-unknown-pragmas -Wno-unknown-warning-option -fPIC -fno-omit-frame-pointer -fvisibility-inlines-hidden -fno-exceptions -fno-asynchronous-unwind-tables -Wgnu-alignof-expression -Wno-gnu-anonymous-struct -Wno-gnu-designator -Wno-gnu-zero-variadic-macro-arguments -Wno-gnu-statement-expression -gdwarf-2 -g3 -O3 -Werror -pedantic -fopenmp=libomp -std=c++11 -o CMakeFiles/rellic-decomp-7.0.dir/rellic/AST/Util.cpp.o -c /var/tmp/xchalup4/rellic/rellic/AST/Util.cpp
/var/tmp/xchalup4/rellic/rellic/AST/ExprCombine.cpp:109:17: error: no matching function for call to 'ignoringParens'
            has(ignoringParens(binaryOperator(stmt().bind("binop")))))) {}
                ^~~~~~~~~~~~~~
/var/tmp/xchalup4/rellic/rellic-build/libraries/llvm/include/clang/ASTMatchers/ASTMatchers.h:795:25: note: candidate function not viable: no known conversion from
      'clang::ast_matchers::internal::BindableMatcher<clang::Stmt>' to 'const internal::Matcher<QualType>' for 1st argument
AST_MATCHER_P(QualType, ignoringParens,
                        ^
/var/tmp/xchalup4/rellic/rellic-build/libraries/llvm/include/clang/ASTMatchers/ASTMatchersMacros.h:130:32: note: expanded from macro 'AST_MATCHER_P'
  AST_MATCHER_P_OVERLOAD(Type, DefineMatcher, ParamType, Param, 0)
                               ^
/var/tmp/xchalup4/rellic/rellic-build/libraries/llvm/include/clang/ASTMatchers/ASTMatchersMacros.h:150:57: note: expanded from macro 'AST_MATCHER_P_OVERLOAD'
  inline ::clang::ast_matchers::internal::Matcher<Type> DefineMatcher(         \
                                                        ^
```

And LLVM 8:

```
[100%] Linking CXX executable rellic-decomp-8.0
/var/tmp/xchalup4/rellic/rellic-build/libraries/cmake/bin/cmake -E cmake_link_script CMakeFiles/rellic-decomp-8.0.dir/link.txt --verbose=1
/var/tmp/xchalup4/rellic/rellic-build/libraries/llvm/bin/clang++  -O2 -g -DNDEBUG  -rdynamic CMakeFiles/rellic-decomp-8.0.dir/rellic/AST/Compat/Stmt.cpp.o CMakeFiles/rellic-decomp-8.0.dir/rellic/AST/Compat/Expr.cpp.o CMakeFiles/rellic-decomp-8.0.dir/rellic/AST/InferenceRule.cpp.o CMakeFiles/rellic-decomp-8.0.dir/rellic/AST/DeadStmtElim.cpp.o CMakeFiles/rellic-decomp-8.0.dir/rellic/AST/CondBasedRefine.cpp.o CMakeFiles/rellic-decomp-8.0.dir/rellic/AST/ExprCombine.cpp.o CMakeFiles/rellic-decomp-8.0.dir/rellic/AST/GenerateAST.cpp.o CMakeFiles/rellic-decomp-8.0.dir/rellic/AST/IRToASTVisitor.cpp.o CMakeFiles/rellic-decomp-8.0.dir/rellic/AST/LoopRefine.cpp.o CMakeFiles/rellic-decomp-8.0.dir/rellic/AST/NestedCondProp.cpp.o CMakeFiles/rellic-decomp-8.0.dir/rellic/AST/NestedScopeCombiner.cpp.o CMakeFiles/rellic-decomp-8.0.dir/rellic/AST/Util.cpp.o CMakeFiles/rellic-decomp-8.0.dir/rellic/AST/Z3CondSimplify.cpp.o CMakeFiles/rellic-decomp-8.0.dir/rellic/AST/Z3ConvVisitor.cpp.o CMakeFiles/rellic-decomp-8.0.dir/rellic/BC/Util.cpp.o CMakeFiles/rellic-decomp-8.0.dir/rellic/BC/Compat/Value.cpp.o CMakeFiles/rellic-decomp-8.0.dir/tools/decomp/Decomp.cpp.o  -o rellic-decomp-8.0 libraries/llvm/lib/libLLVMCore.a libraries/llvm/lib/libLLVMSupport.a libraries/llvm/lib/libLLVMAnalysis.a libraries/llvm/lib/libLLVMipo.a libraries/llvm/lib/libLLVMIRReader.a libraries/llvm/lib/libLLVMBitReader.a libraries/llvm/lib/libLLVMBitWriter.a libraries/llvm/lib/libLLVMTransformUtils.a libraries/llvm/lib/libLLVMScalarOpts.a libraries/llvm/lib/libLLVMLTO.a libraries/llvm/lib/libclangIndex.a libraries/llvm/lib/libclangCodeGen.a libraries/llvm/lib/libclangASTMatchers.a libraries/llvm/lib/libclangTooling.a libraries/z3/bin/libz3.a libraries/glog/lib/libglog.a libraries/gflags/lib/libgflags.a libraries/llvm/lib/libLLVMLTO.a libraries/llvm/lib/libLLVMObjCARCOpts.a libraries/llvm/lib/libLLVMPasses.a libraries/llvm/lib/libLLVMCodeGen.a libraries/llvm/lib/libLLVMTarget.a libraries/llvm/lib/libLLVMCoroutines.a libraries/llvm/lib/libLLVMipo.a libraries/llvm/lib/libLLVMIRReader.a libraries/llvm/lib/libLLVMAsmParser.a libraries/llvm/lib/libLLVMBitWriter.a libraries/llvm/lib/libLLVMInstrumentation.a libraries/llvm/lib/libLLVMLinker.a libraries/llvm/lib/libLLVMVectorize.a libraries/llvm/lib/libLLVMScalarOpts.a libraries/llvm/lib/libLLVMAggressiveInstCombine.a libraries/llvm/lib/libLLVMInstCombine.a libraries/llvm/lib/libLLVMTransformUtils.a libraries/llvm/lib/libLLVMAnalysis.a libraries/llvm/lib/libLLVMCoverage.a libraries/llvm/lib/libLLVMObject.a libraries/llvm/lib/libclangFormat.a libraries/llvm/lib/libclangToolingInclusions.a libraries/llvm/lib/libclangFrontend.a libraries/llvm/lib/libLLVMProfileData.a libraries/llvm/lib/libclangParse.a libraries/llvm/lib/libLLVMMCParser.a libraries/llvm/lib/libclangSerialization.a libraries/llvm/lib/libLLVMBitReader.a libraries/llvm/lib/libclangSema.a libraries/llvm/lib/libclangAnalysis.a libraries/llvm/lib/libclangASTMatchers.a libraries/llvm/lib/libclangEdit.a libraries/llvm/lib/libclangToolingCore.a libraries/llvm/lib/libclangAST.a libraries/llvm/lib/libclangRewrite.a libraries/llvm/lib/libclangLex.a libraries/llvm/lib/libclangDriver.a libraries/llvm/lib/libclangBasic.a libraries/llvm/lib/libLLVMCore.a libraries/llvm/lib/libLLVMMC.a libraries/llvm/lib/libLLVMBinaryFormat.a libraries/llvm/lib/libLLVMDebugInfoCodeView.a libraries/llvm/lib/libLLVMDebugInfoMSF.a libraries/llvm/lib/libLLVMOption.a libraries/llvm/lib/libLLVMSupport.a -lrt -ldl -lm libraries/llvm/lib/libLLVMDemangle.a -lomp -lpthread -lpthread 
/usr/bin/ld: CMakeFiles/rellic-decomp-8.0.dir/rellic/AST/GenerateAST.cpp.o: in function `rellic::GenerateAST::StructureRegion(llvm::Region*)':
/var/tmp/xchalup4/rellic/rellic/AST/GenerateAST.cpp:326: undefined reference to `llvm::RegionBase<llvm::RegionTraits<llvm::Function> >::getNameStr[abi:cxx11]() const'
```

--- a/rellic/AST/ExprCombine.cpp
+++ b/rellic/AST/ExprCombine.cpp
@@ -17,8 +17,8 @@
 #include <gflags/gflags.h>
 #include <glog/logging.h>
 
-#include "rellic/AST/InferenceRule.h"
 #include "rellic/AST/ExprCombine.h"
+#include "rellic/AST/InferenceRule.h"
 #include "rellic/AST/Util.h"
 
 namespace rellic {
@@ -106,7 +106,7 @@ class NegComparisonRule : public InferenceRule {
   NegComparisonRule()
       : InferenceRule(unaryOperator(
             stmt().bind("not"), hasOperatorName("!"),
-            has(ignoringParens(binaryOperator(stmt().bind("binop")))))) {}
+            has(ignoringParenCasts(binaryOperator(stmt().bind("binop")))))) {}
 
   void run(const MatchFinder::MatchResult &result) {
     auto binop = result.Nodes.getNodeAs<clang::BinaryOperator>("binop");

--- a/rellic/AST/IRToASTVisitor.cpp
+++ b/rellic/AST/IRToASTVisitor.cpp
@@ -332,7 +332,7 @@ void IRToASTVisitor::visitGetElementPtrInst(llvm::GetElementPtrInst &inst) {
     auto ptr = inst.getPointerOperand();
     auto fdecl = GetFunctionDecl(&inst);
     std::vector<llvm::Value *> idxs;
-    for (auto &gep_idx : inst.indices()) {
+    for (auto &gep_idx : llvm::make_range(inst.idx_begin(), inst.idx_end())) {
       idxs.push_back(gep_idx.get());
       auto gep_type = llvm::GetElementPtrInst::getGEPReturnType(ptr, idxs);
       auto ref_type = GetQualType(ast_ctx, gep_type);

--- a/scripts/travis.sh
+++ b/scripts/travis.sh
@@ -290,7 +290,7 @@ common_build() {
     ( cd build && make -j `nproc` ) > "${log_file}" 2>&1 &
   else
     printf " i Clang static analyzer enabled\n"
-    ( cd build && scan-build --show-description --status-bugs make -j `GetProcessorCount` ) > "${log_file}" 2>&1 &
+    ( cd build && scan-build --show-description make -j `GetProcessorCount` ) > "${log_file}" 2>&1 &
   fi
 
   local build_pid="$!"

